django/middleware/security.py

```diff
@@ -63,4 +120,22 @@ def process_response(self, request, response):
                 "Cross-Origin-Opener-Policy",
                 self.cross_origin_opener_policy,
             )
+
+        # TODO: Consider if we want to exclude CSP from a default set of views in
+        # DEBUG mode, similar to django-csp.
+        # TODO: Consider if we want to ship with the `@csp_exempt` decorator, to exclude
+        # views from CSP headers.
+        # TODO: Consider if we want to allow per-view configuration of CSP headers via
+        # decorators, similar to django-csp.
```

Comment:
The django-csp project ships with 3 extra decorators, apart from `csp_exempt`:
* `csp` to set a CSP on a view, ignoring what is configured in the Django settings
* `csp_replace` to replace the value of a specified directive
* `csp_update` to add a value to a specified directive
 
I'm not sure if all 3 are fit for Django but I think `csp` would be useful, especially for 3rd party apps to be able to set CSP directives for their own views (e.g. something like wagtail admin, or even Django's own admin).

This does open up security concerns, however, if a 3rd party app has a weaker CSP and potentially allows a XSS attack vector.

Reply:
I decided to go with these sets of decorators, leaving out `csp_replace` and `csp_update` since they seem redundant and also a bit confusing.

To exempt a view from either or both CSP headers:
```python
@csp_exempt(enforced: bool = False, report_only: bool = False)
```

To override the configured Django CSP settings for custom CSP per view, where `policy` is the same as the Django settings:
```python
@csp_enforced(policy: dict[str, Any])
@csp_report_only(policy: dict[str, Any])
```

This highlights 2 different ways we can go with the 2 different CSP dispositions: enforced vs report-only...
1. Specify as method keyword arguments
2. Specify with separate methods

I'm indifferent on which is best but do feel that they should be consistent. Open for discussion on which option is preferred.

Reply:
I ended up circling around to 2 decorators:

1: `csp_exempt`


```python
def csp_exempt(enforced=True, report_only=True)
```

This can be used with or without args. E.g.
```python
@csp_exempt
def view(request):
   return HttpResponse("Exempt both CSP headers")

@csp_exempt(report_only=False)
def view(request):
   return HttpResponse("Enforced header is exempt")
```

2. `csp_override`

```python
def csp_override(config, enforced=True, report_only=True)
```

Usage example:
```python
@csp_override({"DIRECTIVES": {"img-src": [csp.SELF, "data:"]}})
def view(request):
    return HttpResponse("Overrides the policy for both enforced and report-only")

@csp_override({"DIRECTIVES": {"img-src": [csp.SELF, "data:"]}}, enforced=False)
def view(request):
    return HttpResponse("Overrides the policy only for report-only")
```

I think this approach maintains consistency in how they're used -- both decorators accept boolean arguments to configure which CSP header (enforced or report-only) they apply to. By consolidating the previously proposed override decorators into a single decorator, it simplifies usage and avoids the need to remember or stack multiple decorators.